### PR TITLE
:art: no kue dep

### PR DESCRIPTION
**Pull request by KATT**

### CODE REVIEW

File: src/lib/queue.js [hunk 1]
```diff
@@ -1,51 +1,35 @@
...
+
+    // init queue
+    const queue = async.queue(async function (job, callback) {
+        debug('working on job', job);
+        const {_id} = job;
+        try {
+            debug('processRequest')
+            await processRequest(job);
+            debug('setStateAndError')
+            await apiClient.setStateAndError(_id, {state: 'SUCCESS', error: null});
```
Comment (KATT) [inline src/lib/queue.js]: :bug: edge-case if `apiClient.setStateAndError` fails

**Comment by tydicefm** [comment]
:+1:

File: src/index.js [hunk 2]
```diff
@@ -39,54 +27,55 @@ export default function(opts) {
...
-        api  : apiClient,
-        queue: queue,
+        loop,
+        apiClient,
+        queue,
 
-        RequestSchema: RequestSchema,
-        RequestModel : Request,
+        RequestSchema,
+        RequestModel,
```
Comment (SimonRichardson) [inline src/index.js]: :+1:

File: src/lib/schedule-checker.js [hunk 3]
```diff
@@ -38,6 +37,12 @@ export default function(opts) {
         return true;
     }
 
+    function waitFor(ms) {
+        return new Promise((resolve, reject) => {
+            setTimeout(resolve, ms);
+        });
+    }
```
Comment (SimonRichardson) [inline src/lib/schedule-checker.js]: :+1:

**Comment by SimonRichardson** [comment]
:+1:

**Pull request merged**